libs/s25main/figures/nofMiner.h

```diff
@@ -19,10 +20,13 @@ class nofMiner : public nofWorkman
     unsigned short GetCarryID() const override;
     /// Der Arbeiter erzeugt eine Ware
     helpers::OptionalEnum<GoodType> ProduceWare() override;
+    /// alter workcycle (addon)
+    bool isAlteredWorkcycle;
```

Comment:
How is the work cycle "altered" (i.e. "changed", I guess)? I mean: It is the same, it simply produces nothing.
So suggestions: `isFailedMiningAttempt` or `produceNothing` (with a comment: "... in this work cycle")

See also the usage in `ProduceWare` at https://github.com/Return-To-The-Roots/s25client/pull/1501/files#diff-a34581854bed9a5f4ca267f64c0e919f48ffd4b9c09f732351db4c127fb2e932R65-R66 and how the name makes the code more readable: `if(produceNothing) return boost::none`

Reply:
I knew you'd stumble across this :)

I called it altered for two reasons, I may also add an addon which lets you switch between "Produce nothing" and "Produce granite" when using settlers IV mining. This also allows other addons (if one likes) like gold mines have a chance to produce a gem. So in my special case, I could name the variable like you suggested, but as this can be a more general variable I named it ```isAlteredWorkcycle```. Are you good with that, if not, I will change it until this variable becomes another use.